services/121-service/src/payments/reconciliation/intersolve-voucher-reconciliation/intersolve-voucher-reconciliation.service.ts

```diff
@@ -18,11 +18,45 @@ export class IntersolveVoucherReconciliationService {
     private readonly intersolveVoucherScopedRepository: ScopedRepository<IntersolveVoucherEntity>,
   ) {}
 
+  private async updateVouchersWithErrorHandling(
+    vouchers: IntersolveVoucherEntity[],
+    programId: number,
+    credentials: any,
+    errorMessagePrefix: string,
```

Comment:
If you change this to `activityDescription` with values like `updating all balances via cronjob`, constructing the error becomes something like:

```typescript
`${errorVoucherIds.length} errors occurred while ${activityDescription} for vouchers: ${errorVoucherIds.join(', ')}. Failing job to prevent excessive API calls.`,
```

..which I think is easier to read.